flybywiresim/a32nx
# feat(atsu): Datalink status page and Max uplink delay page

## Summary of Changes
This PR adapts the datalink status page to the Hoppie integration and adapts it to MSN7560 layout.
Additionally are some visualizations in the ATC pages fixed to be MSN7560 compliant
The Max Uplink Delay page is also integrated.

## Screenshots (if necessary)



### Updated non-editable entries in MCDU pages
![grafik](https://user-images.githubusercontent.com/64070348/152848385-6e23c288-faec-469a-bd2c-39e1874293b6.png)
![grafik](https://user-images.githubusercontent.com/64070348/152848419-fe8cb9b3-53ec-42dd-b330-415737e3fa3e.png)
![grafik](https://user-images.githubusercontent.com/64070348/152848448-5816ee8d-a2f7-4420-aef1-ebec59838887.png)

### Fixed visualization issues w.r.t MSN7560
![grafik](https://user-images.githubusercontent.com/64070348/152849062-f37b1e9e-a8bb-4bfb-a203-afd10ae84cce.png)

### Max uplink delay (logged in)
![grafik](https://user-images.githubusercontent.com/64070348/151153664-80f7f48b-34f6-4992-a098-2e1cb50c8cc8.png)

### Max uplink delay (logged off)
![grafik](https://user-images.githubusercontent.com/64070348/151153686-259589bb-767a-496c-aca4-40419822e251.png)

### Datalink status without Hoppie connection
![grafik](https://user-images.githubusercontent.com/64070348/151153629-34cd1235-c3a2-4ec6-9513-5dfb1a72a300.png)

### Datalink status with Hoppie connection
![grafik](https://user-images.githubusercontent.com/64070348/151153487-ce3d99a5-5d2a-415d-b1e7-52276ae2cb5f.png)

### Printed datalink page
![grafik](https://user-images.githubusercontent.com/64070348/151153546-b865627e-35fb-45bb-aab2-b24f64773c1b.png)

## References

A320 FCOM documentation for MSN7560




## Additional context



Discord username (if different from GitHub): Sven [de en]

## Testing instructions

- tests of non-editable 
  - keep INIT-page empty
    - AOC/DEPART REQ should show ------ for the callsign and from/to
    - Atc/Connection status should show ---- for current and next ATC
    - Atc/Notification should show ---- for the callsign
  - fill out INIT-page
    - AOC/DEPART REQ should be filled for callsign and from/to
    - Atc/Connection should be the same as above
    - Atc/Notification should show the callsign
- tests of datalink status
  - open the Datalink Status page
    - VHF3 is supposed to show AOC Only
  - activate Hoppie connection via EFB
    - Datalink Status should show without a page reload that AOC/ATC is shown below VHF3
    - deactivation of Hoppie returns to AOC Only
  - All other com channels are not installed
- test print function of datalink status
- Connection Status page
  - check if a handover from one station to an other shows the following information
    - ACTIVE ATC: station that sent the handover
    - NEXT ATC: the following station
    - After an accepted logon of the next station:
    - ACTIVE ATC: old NEXT ATC
    - NEXT ATC: empty
- MAX UPLINK DELAY in CONNECTION
  - Not logged on: show ---- for ACTIVE ATC
  - Logged on: show station in green letters
  - After a logon or handoff is MAX UPLINK DELAY NONE
  - can be set via [1] (is reset after a station change)
  - not editable max uplink delay if no station is active

[1] 
![grafik](https://user-images.githubusercontent.com/64070348/153752475-a9b3c3e2-c0d0-4fcd-965b-ee47ac57a380.png)



## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
